crates/codegen/parser/src/code_generator.rs

```diff
@@ -258,67 +271,68 @@ impl CodeGenerator {
                     .join("crates/codegen/parser_templates/src/shared/macros.rs"),
             )
             .unwrap();
+
         {
-            let scanner_functions = self.scanner_functions().join("\n\n");
+            // Use `format!` here because the content contains comments, that `quote!` throws away.
+            let content = format!(
+                "
+                use super::language::*;
+
+                {scanning_macros}
+                    
+                impl Language {{
+                    {scanner_functions}
+                }}
+                ",
+                scanner_functions = self.scanner_functions()
+            );
+
             codegen
-                .write_file(
-                    &output_dir.join("scanners.rs"),
-                    &format!(
-                        "
-                        use super::language::*;
-
-                        {scanning_macros}
-                            
-                        impl Language {{
-                            {scanner_functions}
-                        }}
-                        ",
-                    ),
-                )
+                .write_file(&output_dir.join("scanners.rs"), &content)
                 .unwrap();
         }
 
         {
-            let parser_functions = self.parser_functions().join("\n\n");
-            let trivia_functions = quote! {
-                fn optional_leading_trivia(&self, stream: &mut Stream) -> Option<Rc<cst::Node>> {
-                    let save = stream.position();
-                    match self.parse_leading_trivia(stream) {
-                        Fail{ .. } => {
-                            stream.set_position(save);
-                            None
-                        },
-                        Pass{ node, .. } => Some(node),
+            // Use `format!` here because the content contains comments, that `quote!` throws away.
+            let content = format!(
+                "
+                use super::language::*;
+                use super::language::ParseResult::*;
+
+                {scanning_macros}
+                    
+                impl Language {{
+                    {trivia_functions}
+                    {parser_functions}
+                }}
+                ",
+                trivia_functions = quote! {
```

Comment:
should we put this inside a similar `self.trivia_functions()`?

Reply:
No, because it is a constant, whereas the other functions are data-driven